contracts/polygon/strategies/mstable/IdleMStableStrategyPolygon.sol

```diff
@@ -0,0 +1,331 @@
+        if (lastIndexAmount > 0) {
+            uint256 gainPerc = ((mUSDStaked - _lastIndexAmount) * 10**20) / _lastIndexAmount;
+            lastApr = (YEAR / (block.timestamp - lastIndexedTime)) * gainPerc;
+        }
+        lastIndexedTime = block.timestamp;
+        lastIndexAmount = uint256(int256(mUSDStaked) + _amount);
+    }
+
+    /// @notice Internal function to redeem the underlying tokens
+    /// @param _amount Amount of strategy tokens
+    /// @return massetReceived Amount of underlying tokens received
+    function _redeem(uint256 _amount, uint256 _price) internal returns (uint256 massetReceived) {
+        uint256 redeemed = (_amount * _price) / oneToken;
+        _updateApr(-int256(redeemed));
+
+        ISavingsContractV2 _imUSD = imUSD;
+        // mUSD we want back
+        uint256 imUSDToRedeem = _imUSD.underlyingToCredits(redeemed);
+        totalLpTokensStaked -= redeemed;
+
+        _burn(msg.sender, _amount);
+        vault.withdraw(imUSDToRedeem);
+        massetReceived = _imUSD.redeemCredits(imUSDToRedeem);
+        underlyingToken.transfer(msg.sender, massetReceived);
+    }
+
+    /// @notice Function to swap the governance tokens on uniswapV2
+    /// @param minLiquidityTokenToReceive minimun number of tokens to that need to be received
+    /// @return _bal amount of underlyings (mUSD) received
+    function _swapGovTokenOnUniswapAndDepositToVault(uint256 minLiquidityTokenToReceive) internal returns (uint256 _bal) {
+        IERC20Detailed _govToken = IERC20Detailed(govToken);
+        uint256 govTokensToSend = _govToken.balanceOf(address(this));
+        IUniswapV2Router02 _uniswapV2Router02 = uniswapV2Router02;
+
+        _govToken.approve(address(_uniswapV2Router02), govTokensToSend);
+        uint256[] memory amounts = _uniswapV2Router02.swapExactTokensForTokens(
+            govTokensToSend,
+            minLiquidityTokenToReceive,
+            uniswapRouterPath,
+            address(this),
+            block.timestamp
+        );
+        uint256 amountReceived = amounts[amounts.length - 1];
+
+        address lastToken = uniswapRouterPath[uniswapRouterPath.length - 1];
+        IERC20Detailed(lastToken).safeApprove(address(underlyingToken), amountReceived);
+        _bal = IMAsset(address(underlyingToken)).mint(lastToken, amountReceived, 0, address(this));
+        _depositToVault(_bal, false);
+        // save the block in which rewards are swapped and the amount
+        latestHarvestBlock = block.number;
+        totalLpTokensLocked = _bal;
+    }
+
+    /// @notice Claim governance tokens
+    /// @param minTokenToClaim min token to be claimed
+    function claimGovernanceTokens(uint256 minTokenToClaim) external onlyOwner {
+        _claimGovernanceTokens(minTokenToClaim);
+    }
+
+    /// @notice Claim governance tokens
+    /// @param minTokenToClaim min token to be claimed
+    function _claimGovernanceTokens(uint256 minTokenToClaim) internal returns (uint256) {
```

Comment:
Can you give me more info on why the changes on `_claimGovernanceTokens` are needed?

Reply:
function signatures are different ethereum and polygon

Reply:
ok but I think we can then simply remove `minTokenToClaim` from here and remove the check `if (rewards + platformRewards > minTokenToClaim)`.
Also do they distribute both MTA and wMATIC? Do you know why there are both `rewards` and `platformRewards`

Reply:
There docs say, but I was not able to find the wMATIC rewards. 
I staked my DAI to confirm that. 


I am looking into it.